### DISCUSSION TITLE
Announcing shadcn-admin v1: What's Coming Next!

### DISCUSSION THREAD

**Discussion by satnaing** [satnaing/shadcn-admin#51] [Announcements]
Hello, everyone! 👋

First of all, I want to extend a huge thank you to everyone who has shown interest in shadcn-admin. 🙌

As some of you may know, shadcn-admin is currently in its beta phase, and I'm excited to share some major updates that are on the way. With the next release, I’m planning a significant version bump to bring even more features, stability, and improvements. Here’s a sneak peek of what you can expect:

### ✨ What's Coming Next

- **_Migrating from React Router to TanStack Router_**: This change will bring better DX and more advanced routing capabilities, making your experience even smoother.
- **_Updating the folder structure to feature-based approach_**: The entire folder structure will be updated to a feature-based organization for better scalability and maintainability.
- **_Replacing the custom sidebar with Shadcn UI's native sidebar_**: To enhance consistency and usability, the new sidebar will leverage Shadcn UI's native components, offering a more seamless design.
- **_Upgrading ESLint from v8 to the Latest v9 Flat Config_**: Expect improved linting capabilities with the latest ESLint updates, helping maintain clean, efficient, and error-free code.

These changes are part of my ongoing effort to make shadcn-admin a robust and scalable admin dashboard solution for everyone.

Stay tuned for more updates, and as always, your feedback is welcome! 🚀💙

Best,
Sat Naing

**Comment by devshinthant** [comment]
Excellent Idea!! 🚀

**Comment by zhuxining** [comment]
why not react router 7?

**Reply by satnaing** [reply]
I've used both [react-router](https://reactrouter.com/en/main) and [tanstack-router](https://tanstack.com/router/latest), and don't get me wrong_ react-router is a fantastic routing library. However, the developer experience (DX) of integrating `tanstack-router` into my React projects is simply better and more intuitive. This might be subjective, of course, but the TanStack ecosystem, combined with better type safety and an improved DX, is arguably superior.

Additionally, this project uses [shadcn-ui](https://ui.shadcn.com/) and `shadcn-ui` itself is leveraging [tanstack-table](https://tanstack.com/table/latest) for its table UI/layouts. So, it just makes sense to use `tanstack-router` for routing as well.

Also, since this is a SPA project, we'll eventually have to deal with some sort of client-side fetching and async state management. I would choose tanstack-query for that too. So, we can have the seamless exp by using`tanstack-query` + `tanstack-router` + `tanstack-table` in one project.

**Comment by satnaing** [comment]
PR https://github.com/satnaing/shadcn-admin/pull/52
